### PR TITLE
docs(bodiless-documentation): Add Shadowing Tutorial

### LINKED ISSUE #2061: [DOC] Shadowing Tutorial
## User Story

AS A developer,
I WANT a shadowing tutorial explaining how to use this technique,
SO THAT I can use this functional tool to reduce development effort.

### Acceptance Criteria

1. The Documentation is updated to include a comprehensive tutorial on using shadowing.

### PR DESCRIPTION
**Pull request by misterjones**
## Changes

- Add "Shadowing Tokens" page.
  - Update ToC with page.
- Remove old Shadowing guide.
  - Update all links to old guide to point to new guide.

## Related Issues

Fixes #2061

### CODE REVIEW

File: packages/vital-examples/doc/Guides/ShadowingTokens.md [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+  </details>
+
+</div>
+
+## Shadowing a Token Collection
+
+To export a shadowed version of a token collection:
+
+01. Add a module to your package which defines the shadowed token collection. You may import the
+    original base token collection to extend it.
```
Comment (wodenx) [inline packages/vital-examples/doc/Guides/ShadowingTokens.md:289]: You are conflating 2 steps here
1. yo uhave to create the actual token collection (in /src/components/Foo) -- as is described in the component template, being sure to extend from the ...Base version of the upstream token collection.
2. you have to add this src/shadow/{base-package}/Foo.ts which imports your token collection and re-exports it (as in this example code).

Comment (misterjones) [reply packages/vital-examples/doc/Guides/ShadowingTokens.md]: @wodenx Rephrased step. ✅ 

Also double-checked previous section ("Creating a Shadowable Token Collection") to ensure that the user was instructed to extend the base version.

File: packages/vital-examples/doc/Guides/ShadowingTokens.md [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+a
+[`vital-demo`](https://github.com/johnsonandjohnson/Bodiless-JS/tree/main/packages/vital-demo/src/shadow/%40bodiless
+':target=_blank') package that shadows all the available Vital components and provides examples. To
+continue using `vital-editors` as our example, if you look in
+[`/packages/vital-demo/src/shadow/@bodiless/vital-editors`](https://github.com/johnsonandjohnson/Bodiless-JS/tree/main/packages/vital-demo/src/shadow/%40bodiless/vital-editors
+':target=_blank'), you can see how to go about shadowing the `EditorPlain` and `RichText`
+components.
+
+The [API documentation](../../Development/API/) is a good resource to use to see what token
+collections are shadowable.
```
Comment (wodenx) [inline packages/vital-examples/doc/Guides/ShadowingTokens.md:93]:
```suggestion
The [API documentation](../../Development/API/) is a good resource to use to see what token
collections are shadowable.
```

File: packages/vital-examples/doc/Guides/ShadowingTokens.md [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+
+To get into the finer details of the structure required to make a component shadowable, please see
+the [Creating a Shadowable Token Collection](#creating-a-shadowable-token-collection) section below,
+where we walk you through the process of structuring your own components to be shadowable.
+
+?> **Note:** The NextJS and GatsbyJS Vital site templates come packaged with the [Vital Design
+System](../), as well as the necessary `tokenShadowPlugin` (see [Shadowing a Token
+Collection](#shadowing-a-token-collection) below), providing a set of shadowable components out of
+the box. You can create a site using one of these templates via the `@bodiless/cli new` command (for
+details, see: [Creating a New Site](../../About/GettingStarted#creating-a-new-site)).
```
Comment (wodenx) [inline packages/vital-examples/doc/Guides/ShadowingTokens.md:103]:
```suggestion
details, see: [Creating a New Site](../../About/GettingStarted#creating-a-new-site)). You can
also use the [component scaffolding tool](./ComponentTemplate.md) to create a shadowable
component in your own package.

```

**Review by wodenx — APPROVED** [review]

**Review by VancheeZze — APPROVED** [review]

**Pull request merged**